lib/elixir/lib/kernel.ex

```diff
@@ -2186,7 +2186,12 @@ defmodule Kernel do
           _          -> arg
         end
 
-        quote do: Module.put_attribute(__MODULE__, unquote(name), unquote(arg))
+        cond do
+          :lists.member(name, [:moduledoc, :typedoc, :doc]) ->
+            quote do: Module.put_attribute(__MODULE__, unquote(name), unquote(arg), __ENV__)
```

Comment:
Pass just the stacktrace instead of the whole env. Like here:

https://github.com/danielberkompas/elixir/blob/3707-warn-on-redefined-docs/lib/elixir/lib/kernel.ex#L2200-L2204